src/routes/bounce_activities_page.jsx

```diff
@@ -0,0 +1,228 @@
+      pagesToDisplay: 5,
+      filterOptions: [],
+      invalidFilter: false,
+      isFetching: true,
+    };
+    this.logout = this.logout.bind(this);
+    this.updateSearchToken = this.updateSearchToken.bind(this);
+    this.updateSearchCategory = this.updateSearchCategory.bind(this);
+    this.addFilter = this.addFilter.bind(this);
+    this.removeFilter = this.removeFilter.bind(this);
+    this.handleModalClose = this.handleModalClose.bind(this);
+    this.handleActivityTabClicked = this.handleActivityTabClicked.bind(this);
+    this.handleBounceTabClicked = this.handleBounceTabClicked.bind(this);
+    this.updatePageIndex = this.updatePageIndex.bind(this);
+    this.handleActivityClicked = this.handleActivityClicked.bind(this);
+    this.handlePrevClicked = this.handlePrevClicked.bind(this);
+    this.handleNextClicked = this.handleNextClicked.bind(this);
+  }
+
+  async componentDidMount() {
+    const { data: activities } = await getActivityLog();
+    if (activities) {
+      this.setState({
+        isFetching: false,
+        activityLog: activities.reverse(),
+      });
+    }
+  }
+
+  logout() {
+    const { history } = this.props;
+    localStorage.clear();
+    history.push("/");
+  }
+
+  handleModalClose(e) {
+    const { id } = e.currentTarget;
+    this.setState({
+      [id]: false,
+      isInvalidInput: false,
+      selectedRule: {},
+    });
+  }
+
+  updateSearchToken(e) {
+    this.setState({
+      searchToken: e.target.value.toLowerCase(),
+    });
+  }
+
+  updateSearchCategory(e) {
+    this.setState({
+      searchCategory: e.value.toLowerCase(),
+    });
+  }
+
+  filterRules(rules) {
+    const { searchToken } = this.state;
+    return rules.filter(
+      rule =>
+        rule.bounce_action.toLowerCase().includes(searchToken) ||
+        rule.description.toLowerCase().includes(searchToken)
+    );
+  }
+
+  paginate(activityLog) {
+    const { currentActivityPageIndex, rulesToShow } = this.state;
+    const ruleStartIndex = (currentActivityPageIndex - 1) * rulesToShow;
+    const ruleEndIndex =
+      (currentActivityPageIndex - 1 * currentActivityPageIndex + rulesToShow) *
+      currentActivityPageIndex;
+    return activityLog.slice(ruleStartIndex, ruleEndIndex);
+  }
+
+  updatePageIndex(e) {
+    const newIndex = parseInt(e.currentTarget.getAttribute("value"), 10);
+    this.setState(prevState => {
+      const isPageIndexUpdated =
+        prevState.currentActivityPageIndex !== newIndex;
+      return {
+        currentActivityPageIndex: isPageIndexUpdated
+          ? newIndex
+          : prevState.currentActivityPageIndex,
+      };
+    });
+  }
+
+  handlePrevClicked() {
+    this.setState(prevState => ({
+      currentActivityPageIndex: prevState.currentActivityPageIndex - 1,
+    }));
+  }
+
+  handleNextClicked() {
+    this.setState(prevState => ({
+      currentActivityPageIndex: prevState.currentActivityPageIndex + 1,
+    }));
+  }
+
+  isDuplicate(searchCategory, searchToken) {
```

Comment:
this can also be written using `some`, which returns true if at least one element in the array passes the test https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some

```
const isDuplicate = (filterOptions, searchValues) => filterOptions.some((filterOption) => {
    return searchValues.searchCategory === filterOption.searchCategory && searchValues.searchToken === filterOption.searchToken
})
```
where searchValues and filterOptions is something like this:
```
const searchValues = {searchCategory: 'date', searchToken: '1'}
const filterOptions = [{searchCategory: 'id', searchToken: '2'}, {searchCategory: 'date', searchToken: '1'}, {searchCategory: 'description', searchToken: '3'}]
```